flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -477,6 +484,18 @@ class TableImpl(
   private def wrap(operation: TableOperation): Table = {
     new TableImpl(tableEnv, operation)
   }
+
+  private[flink] def getOutputFieldReferences: Seq[UnresolvedFieldReferenceExpression] = {
+    operationTree.asInstanceOf[LogicalNode]
+      .output.map(a => new UnresolvedFieldReferenceExpression(a.name))
+  }
+
+  private[flink] def resolveCalls(fields: Seq[Expression]): Seq[Expression] = {
+    val outputFieldReferences = operationTree.asInstanceOf[LogicalNode]
+      .output.map(a => new UnresolvedFieldReferenceExpression(a.name))
+    val expander = new ColumnsOperationExpander(outputFieldReferences)
```

Comment:
Due to the refactor of #8062, the `getOutputFieldReferences()` method will not be used by other places, so I merged `getOutputFieldReferences()` and `resolveCalls()` into one.

Reply:
Why do we need to expands columns here? It seems unnecessary for me. The only reason for the `resolveCalls` in `tableImpl` is because we need to split expressions into `projections`, `aggregates` and `windowProperties`. We don't need to expand columns for that.

Reply:
We need to expand columns before `extractFieldReferences`, so that to get the correct field references.